### PR TITLE
ansible: 2.11.1 -> 2.11.2; 2.10.10 ->2.10.11; 2.9.22 -> 2.9.23

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

New upstream release.

No idea where they keep their release notes or changelog these days. :disappointed: 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
This file looks promising https://github.com/ansible/ansible/blob/stable-2.11/changelogs/CHANGELOG-v2.11.rst#v2-11-2

**Comment by mweinelt** [comment]
Whew, need to rework ansible-collections handling a bit.

File: pkgs/development/python-modules/ansible/collections.nix [hunk 1]
```diff
@@ -14,21 +14,23 @@
 , xmltodict
 , withJunos ? false
 , withNetbox ? false
+
+, version ? null
+, sha256 ? null
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/ansible/collections.nix:19]: I don't think we want the ``? null`` here. If the values are not supplied it should error.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -441,8 +441,6 @@ in {
 
   ansible-base = callPackage ../development/python-modules/ansible/base.nix { };
 
-  ansible-collections = callPackage ../development/python-modules/ansible/collections.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:444]: Do we need an alias for this?

Comment (mweinelt) [reply pkgs/top-level/python-packages.nix:444]: No, ansible should be its only consumer.

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15, nixos x86_64

**Pull request merged**